gtk3/sample/misc/gtkglarea2.rb

```diff
@@ -0,0 +1,213 @@
+=begin
+  gtkglarea2.rb draw a triangle using shaders on a grey background
+  this script need the opengl-bindings gem from :
+  https://github.com/vaiorabbit/ruby-opengl
+
+  gem install opengl-bindings
+
+  sources:
+  http://antongerdelan.net/opengl/hellotriangle.html
+  opengl-bindings/sample/RedBook/varray
+  opengl-bindings/sample/GLES/gles.rb
+  http://www.opengl-tutorial.org/beginners-tutorials/tutorial-2-the-first-triangle/
+  https://developer.gnome.org/gtk3/stable/GtkGLArea.html
+  https://www.bassi.io/articles/2015/02/17/using-opengl-with-gtk/
+  http://stackoverflow.com/questions/30337845/gldrawarrays-not-working-using-gtkglarea-in-gtk3
+=end
+
+require "gtk3"
+require "opengl"
+
+unless Gtk::Version.or_later?(3, 16, 0)
+  puts "This sample requires GTK+ 3.16.0 or later: #{Gtk::Version::STRING}"
+  exit
+end
+
+def getlibdir
+  # http://www.pilotlogic.com/sitejoom/index.php/wiki?id=398<F37>
+  # 32              64
+  # /usr/lib        /usr/lib64       redhat, mandriva
+  # /usr/lib32      /usr/lib64       arch, gento
+  # /usr/lib        /usr/lib64       slackware
+  # /usr/lib/i386.. /usr/libx86_64/  debian
+  libs = Dir.glob("/usr/lib*/libGL.so") # libs in /usr/lib or /usr/lib64 for most distribs
+  libs = Dir.glob("/usr/lib*/*/libGL.so") if libs.size == 0 # debian like
+  if libs.size == 0
+    puts "no libGL.so"
+    exit 1
+  end
+  # Get the same architecture that the runnning ruby
+  if 1.size == 8 # 64 bits
+    File.dirname(libs.grep(/64/)[0])
```

Comment:
`libs[0]` can also be written `libs.first` as follows:

```ruby
irb>  a = [1, 2, 3]
=> [1, 2, 3]
irb> a.first
=> 1
irb> a[0]
=> 1
```

I want to avoid using array index as much as possible in Ruby.